docs/standard/exceptions/best-practices-for-handling-and-using-exceptions-outline.md

```diff
@@ -0,0 +1,91 @@
+		i.	Logging an exception is not handling it. Logging belongs in the global exception handler.
+		ii.	Catching an exception, collecting diagnostic info, and re-throwing is a good practice.
+			1.	If you create a new exception object be sure to set the InnerException property to the original exception.  This ensures you preserve the stack trace.
+		iii.	Properly handling an exception often involves recovery and/or retry.  Your handling logic should be able to perform these tasks if required.
+			1.	Assess the specific type of error you are handling.
+			2.	Verify that your environment is in a known state. 
+			3.	Attempt to cancel or rollback any data transactions.
+		iv.	Handle only the code inside the try block.  If necessary, re-throw the error and let higher level code clean up as required.
+	5.	Shutting the application down or returning to the main method or home page after an exception is often the best choice.
+		i.	An investment in preventing exceptions is preferred to an investment in handling them.
+	6.	Use a finally block to execute code that should run even if an error occurs.		
+		i.	Use a finally block to clean up unmanaged resources or set the environment to a state that calling code expects.
+
+3.	Guidelines for throwing exceptions
+	1.	Throwing exceptions properly is a good practice.
+		i.	Helps identify weaknesses in calling code.
+		ii.	Makes errors easier to find.
+		iii. Aids in expressing and enforcing explicit business logic.
+		iv.	Helps prevent data corruption.
+	2.	Throw exceptions immediately in response to events which have already happened which should have been prevented.
+		i.	An exception should be thrown immediately if code is passed invalid parameters that should have been validated or the application is allowed to  enter an invalid state.
+		ii.	Do not return an error code or null value if business logic has been violated or not enforced when it should have been.  Do not make assumptions on behalf of calling code. Throw an exception immediately.
+	3.	Do not throw exceptions to direct program flow.
+		i.	Avoid throwing an exception to prevent something from happening or to force something to happen that would otherwise not occur. 
+		ii.	Return a null value or error code if code cannot execute as the caller expects but business logic has not been violated.
+			1.	This guideline covers cases where a user might supply an invalid value as a parameter.  In this case the program should display an error message and give the user the opportunity to try again.
+		iii.	Exceptions are expensive and are a means of last resort.
+	4.	Construct a meaningful and helpful error message when throwing an exception.
+		i.	Use proper grammar and check your spelling.
+		ii.	Include as much diagnostic information as possible in your error message.
+		iii. Include instructions on how the caller might prevent the error and/or the location of additional resources to help them resolve it.
+
+4.	Multi-threaded scenarios and Tasks

```

Comment:
This will get very complicated very fast. It's worth having, but recommend a second article devoted to mulithreading and exceptions.